src/runtime/generator.cpp

```diff
@@ -88,6 +111,7 @@ void generatorEntry(BoxedGenerator* g) {
     // we returned from the body of the generator. next/send/throw will notify the caller
     g->entryExited = true;
     threading::popGenerator();
+    freeGeneratorStack(g);
```

Comment:
I think it's because we end up saving the stack in available_addrs, and even if we hit the max number of saved stacks (5) we free a different one (the one that had been in there the longest) to try to encourage cache reuse.  I tried changing it to saving 0 stacks and was able to get a segfault.